python/GafferArnoldTest/ArnoldShaderTest.py

```diff
@@ -52,6 +53,32 @@
 import GafferSceneTest
 import GafferArnold
 
+# Decorator that executes in a subprocess with our test metadata file on the
+# `ARNOLD_PLUGIN_PATH`.
+def withMetadata( func ) :
+
+	def wrapper( self ) :
+
+		metadataPath = os.path.join( os.path.dirname( __file__ ), "metadata" )
+		if metadataPath not in os.environ["ARNOLD_PLUGIN_PATH"].split( ":" ) :
+
+			env = os.environ.copy()
+			env["ARNOLD_PLUGIN_PATH"] = env["ARNOLD_PLUGIN_PATH"] + ":" + metadataPath
+
+			try :
+				subprocess.check_output(
+					[ "gaffer", "test", "GafferArnoldTest.ArnoldShaderTest." + func.__name__ ],
+					env = env, stderr = subprocess.STDOUT
+				)
+			except subprocess.CalledProcessError as e :
+				self.fail( e.output )
+
+		else :
+
+			func( self )
+
+	return wrapper
```

Comment:
I'm wondering if it would be worth the slight performance gain to do this once, and run all 4 of the tests that depend on metadata, rather than starting up a separate subprocess for each of the 4 tests?